### PR TITLE
Add version popout to "Packaged Distributions" table

### PR DESCRIPTION
**Pull request by thesuperzapper**
This PR improves the "Packaged Distributions" table by making it clearer how old/new each Kubeflow Version is. Right now, it's very confusing to new users, as they might not know that Kubeflow 1.7 is actually very old (released in March 2023), so they might start the process of adopting a long-abandoned distribution.

__The key changes this PR makes are:__

1. In the "Packaged Distributions" table under the "Kubeflow Version" column:
     - Put a hover-able icon that says how long ago that version of Kubeflow was released. (Note, this is the _Kubeflow_ released date, not the release date of the distribution).
     - ~~Highlight distributions which are up to date (within 1 minor version of the latest), and de-emphasize distributions which are 2+ minor versions out of date.~~
     - ~~Place warning icons beside versions which are very old (2 = warning, 3+ = scull)~~
2. Remove the very old link to the historical "conformance program" which was started by google but never continued.

__Please note, all of the styling is automatic:__

- when the short-codes for that distribution's version are updated everything will change __automatically__
- when we release a new version of Kubeflow, can use the `content/en/docs/started/installing-kubeflow/get_new_releases.sh` to cache that version's release date into the `content/en/docs/started/installing-kubeflow/release-info` folder of JSON (the build will fail to remind you if you try referencing a version which it does not know about)

# Screenshots

## OLD

Screenshot 2024-10-14 at 12 10 41

### Table Overview

Screenshot 2024-10-23 at 11 05 21

### On-Hover ("Latest Version")

Screenshot 2024-10-29 at 12 29 50

### On-Hover ("1 from Latest Version")

Screenshot 2024-10-29 at 12 29 20

### On-Hover ("3 from Latest Version")

Screenshot 2024-10-29 at 12 30 11

### CODE REVIEW

**Comment by thesuperzapper** [comment]
@jbottum @andreyvelich @kimwnasptd @terrytangyuan @ederign please take a look at this important style update to the "Packaged Distributions" table.

It should help avoid confusion with new users about which distributions are currently supported.
See the preview site here:

- https://deploy-preview-3842--competent-brattain-de2d6d.netlify.app/docs/started/installing-kubeflow/#kubeflow-platform

**Comment by juliusvonkohout** [comment]
The screenshots look good to me.

**Comment by jbottum** [comment]
Since this impacts how distributions are promoted, my suggestion is that it would be good if a majority of the distribution owners approved this i.e. we need 6 distributions to provide a +1  or lgtm

**Comment by jbottum** [comment]
I personally like the version, but am -1 on the emojis.   I propose that this should be implemented if a majority of the distributions are +1.

**Comment by thesuperzapper** [comment]
@jbottum @terrytangyuan as discussed in the last community call, I have removed the warning icons (see new screenshots), which should make this PR very non-controversial now.

As agreed in that meeting, we will give people a week or so to make comments about the proposed change (I will highlight it again in the community meeting tomorrow) and if we don't hear anything, I think we can go ahead and merge this.

_PS: the code will automatically keep everything up to date, so there is no maintenance overhead to this PR_

File: layouts/shortcodes/kf-version-notice.html [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+{{- else if eq $this_version_lag 2 }}
+  {{- $button_style = "btn-kf-version-legacy" }}
+  {{- $popup_icon = "" }}
+  {{- $popup_title = "Version Warning" }}
+  {{- $popup_content = printf `%s<br><br>You may wish to confirm this distribution is still supported by its maintainers.` $popup_content_prefix }}
+{{- else }}
+  {{- $button_style = "btn-kf-version-legacy" }}
+  {{- $popup_icon = "" }}
+  {{- $popup_title = "Version Warning" }}
+  {{- $popup_content = printf `%s<br><br>This distribution may no longer be supported by its maintainers.` $popup_content_prefix }}
```
Comment (tarilabs) [inline layouts/shortcodes/kf-version-notice.html:52]:
```suggestion
  {{- $popup_content = printf `%s<br><br>We recommend for you to confirm this distribution is still supported by its maintainers.` $popup_content_prefix }}
```

**Review by tarilabs — COMMENTED** [review]
Based on the understanding from the community call 2024-10-15, I have some doubt about the use of grey colour, since as well I understand (from the same call) there is not in place an automated conformance-programme, and the rules by which it moves from blue colour to grey colour.
Hence, I wonder what about instead of grey, the table cell in that case is simply rendered as-before.
i.e.:
- blue pill for "recent" in blue, imho ok 
- do not grey pill
- on-hover, imho ok

Personally, I find the on-hover very helpful.
I have one change suggestion however, because I think the wording on the last on-hover is a bit "assuming".

_“Opinions expressed are my own.”_

File: layouts/shortcodes/kf-version-notice.html (hunk 1 above)
Comment (thesuperzapper) [reply layouts/shortcodes/kf-version-notice.html]: Done.

**Comment by thesuperzapper** [comment]
@andreyvelich I have updated the PR with the changes discussed from the community meeting (all distributions are now blue, and the wording is less strong on the 3+ versions out of date message).

I think you can approve this now.

**Comment by jbottum** [comment]
@thesuperzapper thanks for these updates.    I am lukewarm to the commentary in the version popups i.e. the 2nd sentences.   The 1st sentence already tells the user if the distribution is current.    Do we need the 2nd sentence ?  Do they need to be different if the distribution is not current ?   If not current, then something like...Please check with this distribution's maintainers to understand their support plans.

**Comment by thesuperzapper** [comment]
@jbottum I have updated the text in the second paragraph so that they all have the same text.

See the updated screenshots above, or deploy preview.

If no one has any disagreements by the next community meeting, we should be good to merge this.

**Comment by jbottum** [comment]
@nvkassus @chensun @james-jwu @yhwang @johnugeorge @nagar-ajay @rimolive @juliusvonkohout @xujinheng @alexeadem please review the proposed presentation of your distribution on the Installing Kubeflow page and provide your +1 or -1 and/or comments.   We have put a lot of time/work into this (during the Community calls) and would like to finish up and merge.   thanks!

File: content/en/docs/started/installing-kubeflow/release-info/latest.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (andreyvelich) [inline content/en/docs/started/installing-kubeflow/release-info/latest.json:1]: We might need to add information on how to update those JSONs into release Handbook: https://github.com/kubeflow/community/blob/master/releases/handbook.md
cc @kubeflow/release-team 

@thesuperzapper Would it be easier to have single JSON with the list of Kubeflow versions and publish dates ?

Comment (thesuperzapper) [reply content/en/docs/started/installing-kubeflow/release-info/latest.json:1]: @andreyvelich you can add that if you like, but the system is designed to fail the website build when a distribution tries to claim a version that it does not know about.

The error will tell the person making that PR to run the script at [`./content/en/docs/started/installing-kubeflow/get_new_releases.sh`](https://github.com/kubeflow/website/blob/91fa5f5a53df057166c1bee5733914d49a2f216e/content/en/docs/started/installing-kubeflow/get_new_releases.sh).

That script uses the [releases on the `kubeflow/manfistes`](https://github.com/kubeflow/manifests/releases) repo to get the versions and determine the release dates.

Comment (thesuperzapper) [reply content/en/docs/started/installing-kubeflow/release-info/latest.json:1]: But you have reminded me to run the script on this PR as we just released 1.9.1, I will do that now.

File: content/en/docs/started/installing-kubeflow/release-info/v1.6.0.json [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{
```
[the remark this answers is not in the document]
Comment (thesuperzapper) [reply content/en/docs/started/installing-kubeflow/release-info/v1.6.0.json:1]: It's just easier for the automated script to get all of them, and it's not like there are that many.

**Comment by alexeadem** [comment]
> @nvkassus @chensun @james-jwu @yhwang @johnugeorge @nagar-ajay @rimolive @juliusvonkohout @xujinheng @alexeadem please review the proposed presentation of your distribution on the Installing Kubeflow page and provide your +1 or -1 and/or comments. We have put a lot of time/work into this (during the Community calls) and would like to finish up and merge. thanks!

Can we fix the documentation link for the Kubeflow/QBO distro it should be https://www.qbo.io/demos/kubeflow 
We made a couple of changes to our website.  If a PR is needed please advise where to submit it

**Comment by andreyvelich** [comment]
Thank you for this effort @thesuperzapper!
/lgtm
/assign @kubeflow/wg-training-leads @kubeflow/wg-pipeline-leads @kubeflow/wg-data-leads @kubeflow/wg-notebooks-leads @kubeflow/wg-manifests-leads

**Comment by thesuperzapper** [comment]
@james-jwu or @andreyvelich I think we can approve this now, as agreed in the community meeting.

**Comment by thesuperzapper** [comment]
@jbottum can you please approve this, rather than just lgtm?

**Pull request merged**